src/bot.coffee

```diff
@@ -0,0 +1,143 @@
+{Adapter, TextMessage, EnterMessage, LeaveMessage, TopicMessage, Message} = require.main.require 'hubot'
```

Comment:
The _primary_ cause of the Hubot Dev Dependency breaking is actually the calls to `instanceof` in the old Listener.coffee. By eliminating those, the dev dependency issue has basically gone away, because now you can have two identical installs, and everything _should_ be OK. I'm not honestly sure what `require.main.require` means, though.